lib/appsignal.rb

```diff
@@ -305,6 +305,15 @@ def const_missing(name)
             "Please update the constant name to Appsignal::Probes " \
             "in the following file to remove this message.\n#{callers.first}"
         Appsignal::Probes
+      when :Heartbeat
+        unless @heartbeat_constant_deprecation_warning_emitted
```

Comment:
I would expect this not to be necessary as the Heartbeat constant is only fetched once.

Reply:
Is it not fetched each time, if someone, for example, `Appsignal::Heartbeat.new()` several times? I don't really know how these work.

Reply:
Oh yeah, then it would. It doesn't alias the returned value to the constant.

Our current implementation just keeps logging the same warning. Which I've heard no one complain about 😬 

```
$ cd appsignal-ruby
$ bundle exec rake console
irb(main):001> Appsignal::Minutely
appsignal WARNING: The constant Appsignal::Minutely has been deprecated. Please update the constant name to Appsignal::Probes in the following file to remove this message.
(irb):1:in `<top (required)>'
=> Appsignal::Probes
irb(main):002> Appsignal::Minutely
appsignal WARNING: The constant Appsignal::Minutely has been deprecated. Please update the constant name to Appsignal::Probes in the following file to remove this message.
(irb):2:in `<top (required)>'
=> Appsignal::Probes
irb(main):003> Appsignal::Minutely
appsignal WARNING: The constant Appsignal::Minutely has been deprecated. Please update the constant name to Appsignal::Probes in the following file to remove this message.
(irb):3:in `<top (required)>'
=> Appsignal::Probes
```

Reply:
It's probably not an issue in practice because customers call `Appsignal::Minutely` a finite number of times, they don't register probes dynamically.